docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,101 @@
+---
+title: add-discovery-kernel-arguments
+authors:
+- "@oamizur"
+  creation-date: 2022-09-20
+  last-updated: 2022-10-23
+---
+
+# Add discovery kernel arguments
+
+## Summary
+
+Some environments require additional kernel arguments to boot host machines.  
+We would like to support additional kernel arguments while booting the discovery image.
+
+## Motivation
+
+Previously, adding kernel arguments was required by several customers in order to adapt to various environments.
+For example **rd.net.timeout.carrier** may be required in order for either static networking to be applied or 
+to successfully get a dhcp address before pulling the rootfs.
+### Goals
+Add capability to set additional kernel arguments to discovery image for append.
+
+- Image types: Full ISO, minimal ISO, iPXE
+- Deployment types: Cloud, operator
+
+API which supports full set of operations as defined by Coreos Installer: append, replace, delete.
+
+### Non-Goals
+Manipulation of existing kernel parameters (replace/delete).  Replace and delete of exiting
```

Comment:
Given that this is now a goal it shouldn't be a non-goal.

Reply:
Agree, they can be implemented as separate tasks if we choose but can be covered by the same enhancement.

Reply:
@avishayt you indicated that the API needs to be stable but the backend implementation may change later.  Therefore, this is still non-goal.

Reply:
It's a non-goal for the current task that you're implementing.  But it is a goal of this enhancement that should cover delete and replace as well.

Reply:
Changed